src/main/java/de/refactoringbot/refactoring/supportedrefactorings/RemoveMethodParameter.java

```diff
@@ -30,299 +34,298 @@
-		// Check if Method has body
-		if (method.getBody().isPresent()) {
-			// Get body
-			BlockStmt methodBody = method.getBody().get();
-			// Get all expressions from body
-			List<NameExpr> expressions = methodBody.findAll(NameExpr.class);
-			// Iterate expressions
+	private boolean isParameterUsed(MethodDeclaration method, String paramName) {
+		Optional<BlockStmt> methodBody = method.getBody();
+		if (methodBody.isPresent()) {
+			List<NameExpr> expressions = methodBody.get().findAll(NameExpr.class);
 			for (NameExpr expression : expressions) {
-				// If expression name = param name
 				if (expression.getNameAsString().equals(paramName)) {
-					// Parameter used
 					return true;
 				}
 			}
 		}
 
-		// Parameter not used
 		return false;
 	}
 
 	/**
-	 * This method gets the Method signature that our methods will have without the
-	 * parameter.
+	 * Check if post refactoring signature already exists in given class or
+	 * interface
 	 * 
-	 * @param refactoring
-	 * @param paramName
-	 * @return signature
-	 * @throws FileNotFoundException
+	 * @param currentClassOrInterface
+	 * @param postRefactoringSignature
 	 * @throws BotRefactoringException
 	 */
-	private String getPostRefactoringSignature(ParserRefactoring refactoring, String paramName)
-			throws FileNotFoundException, BotRefactoringException {
-		for (String javaFile : refactoring.getJavaFiles()) {
-			// Create compilation unit
-			FileInputStream methodPath = new FileInputStream(javaFile);
-			CompilationUnit compilationUnit = LexicalPreservingPrinter.setup(JavaParser.parse(methodPath));
-
-			// Get all Methods and MethodCalls of File
-			List<MethodDeclaration> fileMethods = compilationUnit.findAll(MethodDeclaration.class);
-
-			// Rename all Methods
-			for (MethodDeclaration fileMethod : fileMethods) {
-				if (refactoring.getMethods().contains(fileMethod)) {
-					performRemoveMethodParameter(fileMethod, paramName);
-					return RefactoringHelper.getMethodSignatureAsString(fileMethod);
+	private void validateClassOrInterfaceNotContainsPostRefactoringSignatureAlready(
+			ClassOrInterfaceDeclaration currentClassOrInterface, String postRefactoringSignature)
+			throws BotRefactoringException {
+		if (RefactoringHelper.isLocalMethodSignaturePresentInClassOrInterface(currentClassOrInterface,
+				postRefactoringSignature)) {
+			throw new BotRefactoringException(
+					"Removal of parameter would result in method signature already present in class or interface '"
+							+ currentClassOrInterface.getNameAsString() + "'.");
+		}
+	}
+
+	/**
+	 * Removes the parameter from all relevant method declarations and method calls
+	 * in the given java files
+	 * 
+	 * @param javaFilesRelevantForRefactoring
+	 * @param issueFilePath
+	 * @param targetMethod
+	 * @param parameterName
+	 * @throws FileNotFoundException
+	 */
+	private void removeParameterFromRelatedMethodDeclarationsAndMethodCalls(
```

Comment:
In this case I try to be as specific as possible, because method, method declaration, method call etc. refer to different things. I try to follow this scheme throughout the code.